### PR TITLE
use promise instead of observer in oauth authentication

### PR DESCRIPTION
**Pull request by ghiobi**
- upgrade storage drive for the app
- upgrade observable version

### CODE REVIEW

File: src/providers/auth.service.ts [hunk 1]
```diff
@@ -129,12 +125,10 @@ export class AuthService {
   public deleteUser() {
     return new Observable((observer) => {
       this.authHttp.delete(environment.server_url + "/api/user")
-        .subscribe(
-          res => {
+        .subscribe((res) => {
```
Comment (JTalarico) [inline src/providers/auth.service.ts:128]: Good attention to detail. This follows rxjs observable format.

File: src/providers/auth.service.ts [hunk 2]
```diff
@@ -44,25 +44,21 @@ export class AuthService {
...
-                this.user = data.data;
-                observer.next(this.user);
-                observer.complete();
-              });
-              browser.close();
-            }
-          });
+        browser.on("loadstart").subscribe((event) => {
+          if (event.url.indexOf(environment.server_url + "/api/app/callback") === 0) {
+            browser.close();
```
Comment (JTalarico) [inline src/providers/auth.service.ts:53]: Should browser really close this soon? Wouldnt it be better to close later?

File: package.json [hunk 3]
```diff
@@ -24,13 +24,13 @@
...
     "@ionic-native/toast": "^3.4.4",
-    "@ionic/storage": "1.1.7",
+    "@ionic/storage": "^2.0.1",
     "angular2-jwt": "^0.1.27",
     "ionic-angular": "2.2.0",
     "ionic-native": "2.8.1",
     "ionic2-input-mask": "^0.18.6",
     "ionicons": "3.0.0",
-    "rxjs": "5.0.0-beta.12",
+    "rxjs": "^5.3.0",
```
Comment (JTalarico) [inline package.json:33]: I'll make a note on slack to inform everyone they need to run npm update before running the code.

Comment (ghiobi) [reply package.json:33]: yes patron.

**Review by JTalarico — APPROVED** [review]
I've reviewed and ran the code and it works well. This is a sensible change as authentication should only be happening once (therefore observable is unnecessary).

**Pull request merged**